share/wake/lib/core/option.wake

```diff
@@ -98,6 +98,15 @@ export def omapPartial (f: a => Option b): Option a => Option b = match _
     Some a -> f a
     None -> None
 
+# omapFailable: A variant of omap that calls a function on a non-empty option that returns a Result
+# type. The Result-type-ness is then hoist outside the resulting option, making it easier to write
+# `require Pass` statements on code that maps over `Option` types.
+export def omapFailable (fn: a => Result b fail): Option a => Result (Option b) fail = match _
```

Comment:
Give us typeclasses and we can have a nice, unguessable `traverse`! 😜  I think the existing naming pattern would have this be `omapPass`, but I don't like the existing naming pattern so I'm happy with this.  It's definitely a function that I can see getting a lot more use out of now that it's in the main library (and now that I'm reminded it exists).

Reply:
I just copied over from the `utils` dir but happy to rename if you want

Reply:
We probably need to go through the library as a whole and make sure we're using consistent language (and specifically the consistent language we actually want) at some point, so I'd say leave it and we'll update later as necessary.  Besides, this still isn't *exactly* the same semantics as `rmapPass` (which returns a `Fail` if the original value was a `Fail`, while this would make it a `Pass None`).